File: PetShop/PetShop.Model/Narudzba.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PetShop.Model
{
    public class Narudzba
    {
        public int Id { get; set; }
        public bool Aktivna { get; set; }
        public bool Zavrsena { get; set; }
        public DateTime Datum { get; set; }
        public int KorisnikId { get; set; }

        public virtual Korisnik Korisnik { get; set; }
    }
}

File: petshop_mobile/lib/models/Narudzba.dart

class Narudzba {
  final int id;
  final bool? aktivna;
  final bool? zavrsena;
  final DateTime? datum;
  final int? korisnikId;

  Narudzba(
      {required this.id,
        required this.aktivna,
        required this.zavrsena,
        required this.datum,
        required this.korisnikId,
      });

  factory Narudzba.fromJson(Map<String, dynamic> json) {
    return Narudzba(
        id: int.parse(json["id"].toString()),
        aktivna: json["aktivna"],
        zavrsena: json["zavrsena"],
        datum: DateTime.tryParse(json['datum']),
        korisnikId: int.parse(json["korisnikId"].toString()),
        );
  }

  Map<String, dynamic> toJson() => {
    "id": id,
    "aktivna": aktivna,
    "zavrsena": zavrsena,
    "datum":
      datum == null ? null : datum!.toIso8601String(),
    "korisnikId": korisnikId,
  };
}
